lib/tether/toolbox.go

```diff
@@ -201,9 +212,17 @@ func (t *Toolbox) killHelper(session *SessionConfig, name string) error {
 
 	num := syscall.Signal(sig.Signum())
 
-	log.Infof("sending signal %s (%d) to process group for %s", sig.Signal, num, session.ID)
-	if err := syscall.Kill(-session.Cmd.Process.Pid, num); err != nil {
-		return fmt.Errorf("failed to signal %s group: %s", session.ID, err)
+	var pid int
+	if groupKill == false {
+		pid = session.Cmd.Process.Pid
+		log.Infof("sending signal %s (%d) to process %d for %s", sig.Signal, num, pid, session.ID)
+	} else {
+		pid = -session.Cmd.Process.Pid
```

Comment:
Why the negative sign here, but not above? (It seems like there's some significance to the sign of the pid, but I'm not familiar enough with this code to know what it means.)

Reply:
Sending a signal to a positive pid sends the signal to one process only; the one identified by the pid.  Sending a signal to a negative pid sends the signal to all the processes belonging to the process group identified by the positive value of the pid.